### ISSUE TITLE
0.3.0 core module depends on 0.3.0-SNAPSHOT json module

### ISSUE THREAD

**Issue by pawelprazak** [VirtusLab/besom#456]
The CI failed with the following during the release:

```
scala-cli --power compile besom-cats --suppress-experimental-feature-warning
Downloading https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/besom-json_3-0.3.0-SNAPSHOT.pom
Downloaded https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/besom-json_3-0.3.0-SNAPSHOT.pom
Downloading https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/besom-json_3-0.3.0-SNAPSHOT.pom.sha1
Downloaded https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/besom-json_3-0.3.0-SNAPSHOT.pom.sha1
Downloading https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/maven-metadata.xml
Downloaded https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/maven-metadata.xml
Downloading https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/maven-metadata.xml.sha1
Downloaded https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/maven-metadata.xml.sha1
[error]  Error downloading org.virtuslab:besom-json_3:0.3.0-SNAPSHOT
  not found: /home/runner/.ivy2/local/org.virtuslab/besom-json_3/0.3.0-SNAPSHOT/ivys/ivy.xml
  not found: https://repo1.maven.org/maven2/org/virtuslab/besom-json_3/0.3.0-SNAPSHOT/besom-json_3-0.3.0-SNAPSHOT.pom
  not found: /home/runner/.cache/scalacli/local-repo/1.1.0/org.virtuslab/besom-json_3/0.3.0-SNAPSHOT/ivys/ivy.xml
  No fallback URL found
error: Recipe `compile-cats` failed on line 65 with exit code 1
Error: Process completed with exit code 1.
```

What happened:
- during the release one version change slipped through the cracks
- core was released to maven with the snapshot dependency (irreversible)

Workarounds (any of them should solve the issue):
1. add `//> using dep "org.virtuslab::besom-core:0.3.0"` to your `project.scala`
2. add `//> using repository sonatype:snapshots` to your `project.scala`
3. upgrade to the upcoming `0.3.1` when it will be released

How to prevent this from happening:
- the release procedure was updated in two places to catch this type of errors before releasing to maven

**Comment by lbialy** [comment]
tl;dr: 0.3.0 is bork, use 0.3.1+ only, we'll keep this issue open for the lifetime of 0.3.x series (hopefully not a very long time)
